### PR TITLE
Resolver updates

### PR DESCRIPTION
**Pull request by gnodet**
- **Resolver updates**
- **Remove a few other references**

### CODE REVIEW

File: maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/RelocatedArtifact.java [hunk 1]
```diff
@@ -116,17 +116,6 @@ public Artifact setVersion(String version) {
         return new RelocatedArtifact(artifact, groupId, artifactId, classifier, extension, version, message);
     }
 
-    @Deprecated
-    @Override
-    public Artifact setFile(File file) {
```
Comment (cstamas) [inline maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/RelocatedArtifact.java:121]: This will break compat, no?

Comment (cstamas) [reply maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/RelocatedArtifact.java:121]: maven-resolver-provider RelocatedArtifact is present in mvn3 as well, I mean "binary compat"

Comment (gnodet) [reply maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/RelocatedArtifact.java:121]: Mmh, not sure, as a default implementation is provided by `AbstractArtifact`... ?

Comment (gnodet) [reply maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/RelocatedArtifact.java:121]: I've revert the removal of the two `setFile()` methods to be safe.

**Review by cstamas — APPROVED** [review]

**Pull request merged**